metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py

```diff
@@ -24,7 +23,48 @@ class SnowflakeCloudProvider(str, Enum):
     AZURE = "azure"
 
 
-SNOWFLAKE_DEFAULT_CLOUD_REGION_ID = "us-west-2"
+# See https://docs.snowflake.com/en/user-guide/admin-account-identifier.html#region-ids
+# This needs an update if and when snowflake supports new region
+SNOWFLAKE_REGION_CLOUD_REGION_MAPPING = {
+    "aws_us_west_2": (SnowflakeCloudProvider.AWS, "us-west-2"),
+    "aws_us_gov_west_1": (SnowflakeCloudProvider.AWS, "us-gov-west-1"),
```

Comment:
this mapping doesn't seem necessary

could be simplified to something like this
```
if ident.startswith('aws_'):
    provider = SnowflakeCloudProvider.AWS
    region = ident.split('_', 1)[1].replace('_', "-")
elif gcp:
    ...
```

Reply:
Yes, I had considered that, but there are some exceptions, so I preferred this path instead.

"aws_us_east_1_gov": (SnowflakeCloudProvider.AWS, "us-east-1"),
"azure_uksouth": (SnowflakeCloudProvider.AZURE, "uk-south"),

Reply:
imo it'd be better to fill the table with the exceptions but still have the generic logic that works.

that way, when new regions are added (and they'll likely be added under the same naming scheme), we automatically work with it instead of needing to update our code

Reply:
yes, that works for me. I have re-written this piece.